code/datums/storage/storage.dm

```diff
@@ -192,10 +195,11 @@
 
 	parent = new_parent
 	// a few of theses should probably be on the real_location rather than the parent
+	if(insert_on_attack)
+		RegisterSignal(parent, COMSIG_ATOM_ATTACKBY, PROC_REF(on_attackby))
```

Comment:
Should register the signal regardless and then check `insert_on_attack` in the signal proc, so stuff can set it independently and not worry about it